x-pack/plugins/case/server/routes/api/utils.ts

```diff
@@ -69,28 +92,93 @@ export const transformNewCase = ({
 });
 
 type NewCommentArgs = CommentRequest & {
+  associationType: AssociationType;
   createdDate: string;
   email?: string | null;
   full_name?: string | null;
   username?: string | null;
 };
 
+/**
+ * Return the alert IDs from the comment if it is an alert style comment. Otherwise return an empty array.
+ */
+export const getAlertIds = (comment: CommentRequest): string[] => {
+  if (isCommentRequestTypeAlertOrGenAlert(comment)) {
+    return Array.isArray(comment.alertId) ? comment.alertId : [comment.alertId];
+  }
+  return [];
+};
+
+/**
+ * This structure holds the alert IDs and indices found from multiple alert comments
+ */
+export interface AlertInfo {
+  ids: string[];
+  indices: Set<string>;
+}
+
+const accumulateIndicesAndIDs = (comment: CommentAttributes, acc: AlertInfo): AlertInfo => {
+  if (isCommentRequestTypeAlertOrGenAlert(comment)) {
+    acc.ids.push(...getAlertIds(comment));
+    acc.indices.add(comment.index);
+  }
+  return acc;
+};
+
+/**
+ * Builds an AlertInfo object accumulating the alert IDs and indices for the passed in alerts.
+ */
+export const getAlertIndicesAndIDs = (comments: CommentAttributes[] | undefined): AlertInfo => {
```

Comment:
nit: `getAlertIndicesAndIDs` and `getAlertIndicesAndIDsFromSO` are almost identical. They can be combined in one function.

Reply:
👍

Reply:
Apparently I don't even use `getAlertIndicesAndIDsFromSO` anywhere haha 🤦  I'll remove it in the next PR.